### PR TITLE
Ensure that the fields mentioned in samples are configurable.

### PR DESCRIPTION
**Pull request by karthikk92**
Ensure that the fields mentioned in samples are configurable.

Fixed issue:

- PV creation is not processed for csi-powerstore installed using csm-operator

- Added checks to validate secrets.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/613 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

File: pkg/drivers/powerstore.go [hunk 1]
```diff
@@ -44,6 +63,64 @@ func PrecheckPowerStore(ctx context.Context, cr *csmv1.ContainerStorageModule, o
...
+			}
+		}
+	}
+
 	log.Debugw("preCheck", "secrets", len(secrets))
 	return nil
 }
+
+// ModifyPowerstoreCR -
+func ModifyPowerstoreCR(YamlString string, cr csmv1.ContainerStorageModule,  file_type string) string {
```
Comment (rensyct) [inline pkg/drivers/powerstore.go:82]: Please maintain the naming convention for variables

File: pkg/drivers/powerstore.go [hunk 2]
```diff
@@ -44,6 +63,64 @@ func PrecheckPowerStore(ctx context.Context, cr *csmv1.ContainerStorageModule, o
...
+	// Parameters to initialise CR values
+	nodeprefix := ""
+	fcportfilter := ""
+	nfsacls := ""
+	healthmonitor_controller := ""
+	chap := ""
+	healthmonitor_node := ""
```
Comment (rensyct) [inline pkg/drivers/powerstore.go:89]: Same comment as above for all the variables declared from Line No 84-89

**Comment by rensyct** [comment]
@karthikk92 , Please ensure to rebase your branch and also fix the linting issues

File: pkg/drivers/powerstore.go (hunk 1 above)
Comment (karthikk92) [reply pkg/drivers/powerstore.go]: Taken care

File: pkg/drivers/powerstore.go (hunk 2 above)
Comment (karthikk92) [reply pkg/drivers/powerstore.go]: taken care

File: operatorconfig/driverconfig/powerstore/v2.5.0/node.yaml [hunk 3]
```diff
@@ -94,35 +94,35 @@ spec:
...
+              value: <X_CSI_POWERSTORE_ENABLE_CHAP>
             - name: X_CSI_POWERSTORE_CONFIG_PATH
               value: /powerstore-config/config
             - name: X_CSI_POWERSTORE_CONFIG_PARAMS_PATH
               value: /powerstore-config-params/driver-config-params.yaml
             - name: GOPOWERSTORE_DEBUG
               value: "true"
             - name: X_CSI_HEALTH_MONITOR_ENABLED
-              value: "false"
+              value: "<X_CSI_HEALTH_MONITOR_ENABLED>"
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerstore/v2.5.0/node.yaml:117]: @karthikk92 , Do we need quotes here

**Review by rensyct — CHANGES_REQUESTED** [review]

File: operatorconfig/driverconfig/powerstore/v2.5.0/node.yaml (hunk 3 above)
Comment (karthikk92) [reply operatorconfig/driverconfig/powerstore/v2.5.0/node.yaml:117]: taken care

**Comment by karthikk92** [comment]
> @karthikk92 , Please ensure to rebase your branch and also fix the linting issues

taken care

File: operatorconfig/driverconfig/powerstore/v2.5.0/node.yaml [hunk 4]
```diff
@@ -94,26 +94,35 @@ spec:
...
+          volumeMounts:
+            - name: driver-path
+              mountPath: /var/lib/kubelet/plugins/csi-powerstore.dellemc.com
+            - name: csi-path
+              mountPath: /var/lib/kubelet/plugins/kubernetes.io/csi
+              mountPropagation: "Bidirectional"
+            - name: pods-path
+              mountPath: /var/lib/kubelet/pods
               value: "false"
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerstore/v2.5.0/node.yaml:126]: Do we have value for volumeMounts. @kk3 please check

File: operatorconfig/driverconfig/powerstore/v2.5.0/node.yaml [hunk 5]
```diff
@@ -94,26 +94,35 @@ spec:
...
+              value: "<X_CSI_POWERSTORE_ENABLE_CHAP>"
             - name: X_CSI_POWERSTORE_CONFIG_PATH
               value: /powerstore-config/config
             - name: X_CSI_POWERSTORE_CONFIG_PARAMS_PATH
               value: /powerstore-config-params/driver-config-params.yaml
             - name: GOPOWERSTORE_DEBUG
               value: "true"
             - name: X_CSI_HEALTH_MONITOR_ENABLED
+              value: "<X_CSI_HEALTH_MONITOR_ENABLED>"
+          volumeMounts:
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerstore/v2.5.0/node.yaml:118]: @kk3, please check if volumeMounts section is duplicated in this file

**Review by rensyct — CHANGES_REQUESTED** [review]

File: operatorconfig/driverconfig/powerstore/v2.5.0/node.yaml (hunk 5 above)
Comment (karthikk92) [reply operatorconfig/driverconfig/powerstore/v2.5.0/node.yaml:118]: taken care

File: operatorconfig/driverconfig/powerstore/v2.5.0/node.yaml (hunk 4 above)
Comment (karthikk92) [reply operatorconfig/driverconfig/powerstore/v2.5.0/node.yaml]: taken care

**Review by rensyct — APPROVED** [review]

**Review by panigs7 — APPROVED** [review]

**Pull request merged**